### PR TITLE
Split git-util into two packages. Added ability to remove files from an sdk.

### PR DESCRIPTION
**Pull request by TimGebert**
## Proposed changes

Split git util into two separate packages. One to manage git flow, other package, download-util, to manage downloading and modification of sdks.

~Added ability in download-util to use globs in order to remove files from a downloaded sdk.~

Currently breaking the tests in git-util package, will need to look at with @PatrickShaw 

## Types of changes

What types of changes does your code introduce to OpenAPI Platform?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have signed and submitted the appropriate [CLA](../docs/contrib)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/download-util/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  const paths = await getAllFilepathsInDir(sdkDir);
+  for (const path of paths) {
+    const fromPath = path;
+    const relativePath = relative(sdkDir, path);
+    const toPath = join(repoDir, relativePath);
+    await move(fromPath, toPath, { overwrite: true });
+  }
+}
+
+async function deleteAllFilesInLocalRepo(dir) {
```
Comment (PatrickShaw) [inline packages/download-util/src/index.ts]: This should be in the git-util package along with anything else that has to do with repos.

File: packages/download-util/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      );
+
+      for (const glob of fileCleaningGlobs) {
+        const globString = join(sdkDir, glob);
+        const matchedPaths = await globby(globString);
+        logger.verbose(`matched files: ${matchedPaths}`);
+
+        await deletePaths(matchedPaths);
+      }
+
```
Comment (PatrickShaw) [inline packages/download-util/src/index.ts]: This file glob type stuff, along with downloading and extracting the SDKs are the only things that should be in the download util package <- You'll need to take out the associated logic from `migrateSdkIntoLocalRepo` and put it in its own method. 

Anything that has to do with git shouldn't be in this package.

File: packages/download-util/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import { join, relative } from 'path';
+
+import AdmZip from 'adm-zip';
+import { move } from 'fs-extra';
+import globby from 'globby';
+import { listFiles } from 'isomorphic-git';
+
+import { downloadToPath, deletePaths, makeTempDir } from './file/index';
+
+// Note: dir = directory
```
Comment (darvid7) [inline packages/download-util/src/index.ts]: Is this really needed?

File: packages/file-util/src/index.ts [hunk 4]
```diff
@@ -15,25 +16,45 @@ export async function deletePaths(paths) {
...
 }
 
+function streamToPromise(stream: ReadStream): Promise<any> {
+  return new Promise((resolve, reject) => {
+    stream.on('error', err => {
+      stream.destroy();
+      reject(err);
+    });
+    stream.on('finish', resolve);
+    stream.on('error', reject);
```
Comment (calebj0seph) [inline packages/file-util/src/index.ts]: Is there a point to this when you already have an error handler above?

Comment (PatrickShaw) [reply packages/file-util/src/index.ts]: Nope :) good eye

File: packages/file-util/src/index.ts [hunk 5]
```diff
@@ -15,25 +16,45 @@ export async function deletePaths(paths) {
...
+  });
+}
+
+export async function downloadToBuffer(remotePath: string) {
+  const response = await fetch(remotePath);
+  // Pipes the response to a file, wrapped a promise so we can use await
+  return streamToPromise(response.body);
+}
```
Comment (PatrickShaw) [inline packages/file-util/src/index.ts]: I've added this because we're going to need it for the PR that I make for storing files.

**Pull request merged**